openedx/features/philu_courseware/tests/test_competency_assessment_recording.py

```diff
@@ -0,0 +1,90 @@
+import copy
+from django.urls import reverse
+
+from rest_framework.test import APITestCase
+from rest_framework import status
+
+from student.tests.factories import UserFactory
+
+from openedx.features.philu_courseware.constants import COMP_ASSESS_RECORD_SUCCESS_MSG, INVALID_PROBLEM_ID_MSG
+
+INVALID_ASSESSMENT_TYPE = 'This is invalid assessment type'
+INVALID_CORRECTNESS = 'This is invalid correctness'
+NOT_VALID_CHOICE_FORMAT = '"{}" is not a valid choice.'
+
+
+class CompetencyAssessmentRecordTest(APITestCase):
+    def setUp(self):
+        self.user = UserFactory.create(username='testuser', password='testpwd')
+        self.client.force_authenticate(self.user)
+        self.end_point = reverse('record_competency_assessment')
+        self.valid_record = {
+            'problem_id': 'block-v1:PUCIT+IT1+1+type@problem+block@7f1593ef300e4f569e26356b65d3b76b',
+            'problem_text': 'This is a problem',
+            'assessment_type': 'pre',
+            'attempt': 1,
+            'correctness': 'correct',
+            'choice_id': 1,
+            'choice_text': 'This is correct choice',
+            'score': 1
+        }
+        self.invalid_problem_id_record = {
+            'problem_id': 'thisisinvalidproblemid',
+            'problem_text': 'This is a problem',
+            'assessment_type': 'pre',
+            'attempt': 1,
+            'correctness': 'correct',
+            'choice_id': 1,
+            'choice_text': 'This is correct choice',
+            'score': 1
+        }
+        self.invalid_correctness_and_assessment_type_record = {
```

Comment:
Lets move this, inside `test_invalid_correctness_and_assessment_type` test case, since it is required there only. `setUp` is called for each test case.

same for `invalid_problem_id_record`